trompace/queries/document.py

```diff
@@ -59,8 +59,8 @@ def query_document(identifier: str=None, document_name=None, publisher=None, con
     if language and language not in SUPPORTED_LANGUAGES:
         raise UnsupportedLanguageException(language)
 
-    if identifier:
-        return query_create(identifier, QUERY_DIGITALDOCUMENT, document_name, publisher, contributor, creator,
-                                 source, description, language)
+    if all(v is None for v in locals().values()):
```

Comment:
I don't understand what this is supposed to do, can you explain it? `locals()` hides a lot of intent, so it's a good idea to avoid it unless absolutely necessary

Reply:
It's basically to check if any of the input variables is present, by using locals() we can ensure that this continuity remains even if we add more arguments, without having separate checks for each variable.

Reply:
Basically, this checks if all of the inputs are None. If so, it returns the query required.

Reply:
what's the purpose of the if/else here? My feeling is that it's not a good idea to have a different behaviour based on if you have arguments or not, but I can't work out what's happening because I can't find a definition of `query_create`

Reply:
If there are no arguments, it would resort to a query which queries all documents. If there is an arguments, it needs to be passed to the query, unfortunately, sending a blank "()" does not work, hence the two queries.

Reply:
ah, right. How about adding the () around the parameters in `query_create` before adding to the template? That way it could be either an empty string, or a string that is formatted `(param: value)`.

Reply:
Done, both here and for person.